docs/src/examples/DTMP_example1.jl

```diff
@@ -0,0 +1,79 @@
+# # Foundations of Bilevel Programming: Example 8
+# This example is from the book Decomposition Techniques in Mathematical Programming
+# Chapter 7.2, page 281, [url](https://www.springer.com/gp/book/9783540276852)
+
+# Bold point: using upperonly and loweronly, using equality in constraints
```

Comment:
we should point out that: upper/loweronly creates variable only in the proper model and hence might generate slightly smaller reformulated models